pkgs/development/interpreters/racket/default.nix

```diff
@@ -1,209 +1,109 @@
-    # See: https://github.com/NixOS/nixpkgs/issues/114993#issuecomment-812951247
-    ./force-cs-variant.patch
-
-    # The entry point binary $out/bin/racket is codesigned at least once. The
-    # following error is triggered as a result.
-    # (error 'add-ad-hoc-signature "file already has a signature")
-    # We always remove the existing signature then call add-ad-hoc-signature to
-    # circumvent this error.
-    ./force-remove-codesign-then-add.patch
-  ];
-
-  preConfigure =
-    ''
-      unset AR
-      for f in src/lt/configure src/cs/c/configure src/bc/src/string.c; do
-        substituteInPlace "$f" \
-          --replace /usr/bin/uname ${coreutils}/bin/uname \
-          --replace /bin/cp ${coreutils}/bin/cp \
-          --replace /bin/ln ${coreutils}/bin/ln \
-          --replace /bin/rm ${coreutils}/bin/rm \
-          --replace /bin/true ${coreutils}/bin/true
-      done
-
-      # The configure script forces using `libtool -o` as AR on Darwin. But, the
-      # `-o` option is only available from Apple libtool. GNU ar works here.
-      substituteInPlace src/ChezScheme/zlib/configure \
-          --replace 'ARFLAGS="-o"' 'AR=ar; ARFLAGS="rc"'
-
-      mkdir src/build
-      cd src/build
-
-    ''
-    + lib.optionalString stdenv.hostPlatform.isLinux ''
-      gappsWrapperArgs+=("--prefix"   "LD_LIBRARY_PATH" ":" ${libPath})
-      gappsWrapperArgs+=("--set"      "LOCALE_ARCHIVE" "${glibcLocales}/lib/locale/locale-archive")
-    ''
-    + lib.optionalString stdenv.hostPlatform.isDarwin ''
-      gappsWrapperArgs+=("--prefix" "DYLD_LIBRARY_PATH" ":" ${libPath})
-    '';
-
-  preBuild = lib.optionalString stdenv.hostPlatform.isDarwin ''
-    # Cannot set DYLD_LIBRARY_PATH as an attr of this drv, becasue dynamic
-    # linker environment variables like this are purged.
-    # See: https://apple.stackexchange.com/a/212954/167199
+  CoreFoundation,
+  glibcLocales,
 
-    # Make builders feed it to dlopen(...). Do not expose all of $libPath to
-    # DYLD_LIBRARY_PATH as the order of looking up symbols like
-    # `__cg_jpeg_resync_to_restart` will be messed up. Our libJPEG.dyllib
-    # expects it from our libTIFF.dylib, but instead it could not be found from
-    # the system `libTIFF.dylib`. DYLD_FALLBACK_LIBRARY_PATH has its own problem
-    # , too.
-    export DYLD_FALLBACK_LIBRARY_PATH="${libPath}"
-  '';
+  disableDocs ? false,
+}:
 
-  shared = if stdenv.hostPlatform.isDarwin then "dylib" else "shared";
-  configureFlags =
-    [
-      "--enable-${shared}"
-      "--enable-lt=${libtool}/bin/libtool"
-    ]
-    ++ lib.optionals disableDocs [ "--disable-docs" ]
-    ++ lib.optionals stdenv.hostPlatform.isDarwin [
-      "--disable-strip"
-      "--enable-xonx"
+racket-minimal.overrideAttrs (
+  finalAttrs: previousAttrs: {
+    pname = "racket";
+    version = "8.14";
+
+    src = previousAttrs.src.override {
+      hash = "sha256-qrjMDbM27S04KAPHCK1VqV/FKkQ2yRL2FvfEnUhFriw=";
+    };
```

Comment:
Relying on finalAttrs.pname here is really none transparent and hard to understand.